packages/scheduler/lib/scheduler.ts

```diff
@@ -103,16 +100,50 @@ export class Scheduler {
      *         createdToStartedTimeoutSecs: 1,
      *         startedToCompletedTimeoutSecs: 1,
      *         heartbeatTimeoutSecs: 1
-     *     }
      * };
      * const scheduled = await scheduler.immediate(schedulingProps);
      */
-    public async immediate(props: ImmediateProps): Promise<Result<Task>> {
-        const taskProps = {
-            ...props,
-            startsAfter: new Date(),
-            scheduleId: null
-        };
+    public async immediate(props: ImmediateProps | { scheduleName: string }): Promise<Result<Task>> {
```

Comment:
this function should run in a transaction

Reply:
You get search the schedule, then search for a task then insert the task.
A concurrent call could invalidate the schedule or that there is no pending task

Reply:
I don't think a transaction would prevent those scenarios.

Reply:
the transaction should lock the rows and execute two concurrent calls to immediate in a row instead of together, but maybe only if you specify FOR UPDATE. Might be more easily solved with a shared lock indeed. 
Also just thinking out loud, a schedule deletion should cancel the running tasks if possible

Reply:
wrote down a note to address concurrent task scheduling. 
For the schedule deletion, yes I will do that when implementing schedule deletion